templates/terraform/resource.erb

```diff
@@ -199,7 +199,17 @@ func resource<%= resource_name -%>Create(d *schema.ResourceData, meta interface{
     var project string
     if parts := regexp.MustCompile(`projects\/([^\/]+)\/`).FindStringSubmatch(url); parts != nil {
         project = parts[1]
-    }
+    } else {
```

Comment:
As proposed, the control flow is roughly:

```
  if `has_project` # has a direct project
    - use that
  else if `supports_indirect_user_project_override`
    if project in url
      - use that
    else if provider.billing_project is set
      - use that
```

Instead, as discussed above, we want something closer to this:

```
  if provider.billing_project is set
    - use that
  else 
	if `has_project` # has a direct project
      - use that
    else if `supports_indirect_user_project_override`
      - find project in url & use that
```

Code-wise, I'd imagine implementing the control flow as looking something like:

```
  billingProject := ""
  if `has_project` # has a direct project
    billingProject = project
  else if `supports_indirect_user_project_override`
    billingProject = project in url

  if provider.billing_project is set
    billingProject = provider.billing_project
```

My apologies- I get to use pseudocode that mixes templating and Go at will. If you're having trouble, I'll gladly try to help clarify my thoughts.

Reply:
I reworked it. Let me know what you think. Also how do I run the VCRs locally? I don't think it is setting the headers correctly.